### ISSUE TITLE
[accessibility] Consider replacing `aria-pressed` with `aria-expanded` on the attribution control

### ISSUE THREAD

**Issue by Malvoz** [maplibre/maplibre-gl-js#359]
Regarding:

https://github.com/maplibre/maplibre-gl-js/blob/0c6ead362589465ffa14a8feb99b689640c698d3/src/ui/control/attribution_control.js#L102-L110

[`aria-pressed`](https://www.w3.org/TR/wai-aria/#aria-pressed) is similar to [`aria-expanded`](https://www.w3.org/TR/wai-aria/#aria-expandedd) in that it indicates the state of the button, but `aria-expanded` is more appropriate as it specifically:

> Indicates whether the element, or another grouping element it controls, is currently expanded or collapsed.

The distinction is that with `aria-expanded` the (non-visual) user will expect other ("new") content to be available after pressing the button. This is mostly important as there is no focus delegation (which is fine if it's conveyed to the user that new content is available through `aria-expanded`).

<hr>

(The same issue was also reported with MapBox: https://github.com/mapbox/mapbox-gl-js/issues/11034.)

**Comment by Malvoz** [comment]
@astridx that's true, HTML is preferred over ARIA where possible.

For example, markers [should ideally be `<button>` as opposed to `<div role="button">`](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/button/button.html) (https://github.com/maplibre/maplibre-gl-js/issues/356). But such changes would require more effort (to reset the styles of these elements) which I don't want to impose, even though that's usually not very hard to achieve.

**Comment by acalcutt** [comment]
Do you think this aria-pressed with aria-expanded change may have broken the attributions when in full screen?

I am testing the Terrain3d branch that is being worked on here ( https://github.com/prozessor13/maplibre-gl-js/commits/terrain3d ) and notice my attributions are no longer showing when in full screen (or above a width of about 615px). they show properly as a button when the screen is smaller (below 615px), but full screen they don't show at all. Looking at the change log I think it might have started when the main branch with this change was merged in.

Latest version at above 615px, missing attributes
![attrib_missing](https://user-images.githubusercontent.com/3792408/141647828-a718d440-e9be-4b51-90ce-7fb295968cc0.png)

Latest version below 615px, the attributes show up as a button like they should
![attrib_smaller](https://user-images.githubusercontent.com/3792408/141647915-b2c84b1b-647c-4327-948e-6e88bcfdd09f.png)


This is how it looked when it was working on the previous version (which i compiled 10/19) above 615px, all my attribution show as a list at the bottom, until the screen gets below a certain size (around 615px) and it turns into a button

![attrib_working](https://user-images.githubusercontent.com/3792408/141648078-f0ba1efa-038e-4c44-8356-72b9c3a9cf12.png)

**Comment by acalcutt** [comment]
I tested reverting the changes in attribution_control.ts [here](https://github.com/astridx/maplibre-gl-js/commit/0d1d7956484dcdd030574b21096159bc1adfcb83#diff-7489f127e6e44739b4d10d3d5102ca30e47ff444ebba85ea9921e558e9468676) and it does in fact fix my full screen attribution issue

**Comment by Malvoz** [comment]
> Do you think this aria-pressed with aria-expanded change may have broken the attributions when in full screen?

Just noting that the original suggested solution was not used.

Closing this issue as it is no longer relevant per https://github.com/maplibre/maplibre-gl-js/pull/557 (and because you've raised https://github.com/maplibre/maplibre-gl-js/issues/640 for the issue you're seeing).
